src/vuefire.js

```diff
@@ -61,11 +85,7 @@ function bind (vm, key, source) {
   if (!isObject(source)) {
     throw new Error('VueFire: invalid Firebase binding source.')
   }
-  // get the original ref for possible queries
-  var ref = source
-  if (typeof source.ref === 'function') {
```

Comment:
The first check is to make sure things work with the Firebase `2.x.x` SDK where `ref()` is a function on both. So, maybe adding comments will help for posterity sake:

```
if (typeof source.ref === 'function') {
  // ref() is a function in Firebase 2.x.x
  ref = source.ref()
} else (typeof source.ref === 'object') {
  // ref is a read-only property in Firebase 3.x.x
  ref = source.ref
}
```